### PR TITLE
poc: Move test vector generation out of unit tests

### PR DESCRIPTION
**Pull request by cjpatton**
Partially addresses #327.

Remove the TEST_VECTOR and TEST_VECTOR_PATH environment variables and move test vector generation logic into a new tool, `gen_test_vec.py`.

The test vectors are complete with the following exceptions:

1. We currently don't have test vectors for Prio3MultihotCountVec. This VDAF is new as of draft 10, but we neglected to add test vectors for it. This change checks in the test vectors.

2. The test vector for IdpfBBCGGI21 was regenerated after this IDPF was renamed. Also, as of 01291c2c7c7568f8d31d01aa4d3e9df653d6b447, we are generating this with a proper nonce rather than an adhoc binder. To confirm that nothing has changed, you can modify `gen_test_vec_for_idpf()` by replacing Lines

    ```
    nonce = gen_rand(idpf.NONCE_SIZE)
    ```

    with

    ```
    nonce = bytes([0x73,0x6f,0x6d,0x65,0x20,0x6e,0x6f,0x6e,0x63,0x65])
    ```

Also, this change opts in `gen_test_vec.py` and
`plot_prio3_multiproof_robustness.py` into linting and type checking.

### CODE REVIEW

File: poc/gen_test_vec.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+                                      test_vec_instance)
+    with open(filename, 'w') as f:
+        json.dump(test_vec, f, indent=4, sort_keys=True)
+        f.write('\n')
+
+
+# XOF
+
+
+def gen_test_vec_for_xof(cls: type[Xof]) -> None:
```
Comment (cjpatton) [inline poc/gen_test_vec.py:303]: Reviewer note: copied verbatim from the tests directory.

File: poc/vdaf_poc/vdaf.py [hunk 2]
```diff
@@ -19,6 +16,7 @@
 PrepState = TypeVar("PrepState")
 PrepShare = TypeVar("PrepShare")
 PrepMessage = TypeVar("PrepMessage")
+F = TypeVar("F", bound=Field)
```
Comment (cjpatton) [inline poc/vdaf_poc/vdaf.py:19]: Reviewer note: Moved this from below to here. I like the idea of having all type variables for a file in one place.

File: poc/gen_test_vec.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+    gen_test_vec_for_vdaf(
+        vdaf_prio3.Prio3MultihotCountVec(2, 4, 2, 2),
+        None,
+        [[0, 1, 1, 0]],
+        0,
+    )
+
+    # Poplar1
+    tests = [
+        (0, [0, 1]),
```
Comment (divergentdave) [inline poc/gen_test_vec.py:377]: nit: the others use tuples, while this uses a list

**Review by divergentdave — APPROVED** [review]
We should update the command in the README. (and cut out the second command about how to filter which test vectors you generated, since that's no longer possible)

File: poc/gen_test_vec.py (hunk 3 above)
Comment (cjpatton) [reply poc/gen_test_vec.py]: Done.

File: poc/gen_test_vec.py [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+    # Poplar1
+    measurements = [0b1101]
+    tests = [
+        (0, [0, 1]),
+        (1, (0, 1, 2, 3)),
+        (2, (0, 2, 4, 6)),
+        (3, (1, 3, 5, 7, 9, 13, 15)),
+    ]
+    for (level, prefixes) in tests:
+        gen_test_vec_for_vdaf(  # type: ignore
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply poc/gen_test_vec.py]: Done! Thanks :)

**Comment by cjpatton** [comment]
> We should update the command in the README. (and cut out the second command about how to filter which test vectors you generated, since that's no longer possible)

That's odd, I thought I had checked in this change! I guess I didn't. Anyway, done now.

**Pull request merged**